substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -427,10 +427,8 @@ impl<T: Config> Pallet<T> {
 		}
 
 		// disable all offending validators that have been disabled for the whole era
-		for (index, disabled) in <OffendingValidators<T>>::get() {
-			if disabled {
-				T::SessionInterface::disable_validator(index);
-			}
+		for index in <OffendingValidators<T>>::get() {
```

Comment:
On each session validators are disabled again here

Reply:
An alternative would be to removed clearing disabled state on session changes:
https://github.com/paritytech/polkadot-sdk/blob/50390950d87fb745ec5ff46be698cac93113efb7/substrate/frame/session/src/lib.rs#L639-L642